### PR TITLE
cron viewer in admin

### PR DESCRIPTION
**Pull request by DominicWatts**
Cron viewer in admin.  With possibility of further development for management.

### Description
In Magento 1 we always use https://github.com/AOEpeople/Aoe_Scheduler to manage crons.  Generally we use it for debugging such as checking crons run on time and whether they throw any errors but it is handy to have the option to quickly trigger a cron manually out-of-schedule. It's always surprised me there isn't a feature like this included in the core.

I am happy to continue this PR further and develop basic management functions although I'm interested in feedback from the community first.

### Fixed Issues (if relevant)
1. magento/magento2#570: Additional information about cron jobs in admin

### Manual testing scenarios
N/A - additional feature

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@DominicWatts the idea is really good, however I think it should be done not as part of Magento/Backend module (because it's not responsibility of this module), but as part of Magento/Cron module OR as a separate module. BTW let's wait some feedback from Magento Team

File: app/code/Magento/Backend/view/adminhtml/layout/adminhtml_cron_block.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0"?>
```
Comment (ishakhsuvarov) [inline app/code/Magento/Backend/view/adminhtml/layout/adminhtml_cron_block.xml]: It is suggested to use UI Components for the grids in the new code, other approaches are considered legacy.

File: app/code/Magento/Backend/Controller/Adminhtml/Cron.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Backend\Controller\Adminhtml;
+
+use Magento\Backend\App\Action;
+use Magento\Framework\Exception\LocalizedException;
+
+abstract class Cron extends Action
```
Comment (ishakhsuvarov) [inline app/code/Magento/Backend/Controller/Adminhtml/Cron.php]: Is this abstract class really needed here? I don't see it's responsibility.

**Comment by ishakhsuvarov** [comment]
@DominicWatts @ihor-sviziev I agree, that it should not be responsibility of the Backend module. In current state, it looks like it will fit Cron module pretty well.

File: app/code/Magento/Backend/view/adminhtml/layout/adminhtml_cron_block.xml (hunk 1 above)
Comment (DominicWatts) [reply app/code/Magento/Backend/view/adminhtml/layout/adminhtml_cron_block.xml]: Sure.  Have you got an example I can look at?

Comment (ishakhsuvarov) [reply app/code/Magento/Backend/view/adminhtml/layout/adminhtml_cron_block.xml]: You may start with the [DevDocs article](http://devdocs.magento.com/guides/v2.2/ui_comp_guide/components/ui-listing-grid.html) on this subject, however syntax in the article is slightly outdated. 

I consider these examples, from MSI project, pretty well implemented: [Component declaration](https://github.com/magento-engcom/magento2/blob/develop/app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_listing.xml) + [DataProvider for the Component](https://github.com/magento-engcom/magento2/blob/develop/app/code/Magento/Inventory/Ui/DataProvider/SourceDataProvider.php).
Please let us know if you have any more questions regarding this.

**Comment by piotrekkaminski** [comment]
Thanks for contributing. AOE Scheduler was a very popular solution and given that the team was not planning to update it to M2, i would love to see it included.

**Comment by DominicWatts** [comment]
I've implemented the new style grids.  Although having a couple of issues.  Pagination backwards doesn't seem to be working properly and the search and I have no idea why.  Any assistance will be greatly appreciated.  Once the grid is perfected I am happy to continue on and develop some schedule management functionality.

File: app/code/Magento/Cron/view/adminhtml/ui_component/adminhtml_cron_listing.xml [hunk 3]
```diff
@@ -0,0 +1,437 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (omiroshnichenko) [inline app/code/Magento/Cron/view/adminhtml/ui_component/adminhtml_cron_listing.xml:1]: Try to rewrite configuration on new style(e.g https://github.com/magento/magento2/blob/develop/app/code/Magento/Customer/view/adminhtml/ui_component/customer_listing.xml).

**Comment by vkublytskyi** [comment]
@piotrekkaminski, @DominicWatts is this PR still relevant as AOE scheduler for M2 was released:
- https://github.com/AOEpeople/Aoe_Scheduler/issues/94#issuecomment-331202266
- https://www.wyomind.com/magento2/cron-scheduler-magento.html

**Pull request closed without merging**